stubs/requests/requests/sessions.pyi

```diff
@@ -45,8 +45,15 @@ class SessionRedirectMixin:
     def rebuild_proxies(self, prepared_request, proxies): ...
     def should_strip_auth(self, old_url, new_url): ...
 
-_Data: TypeAlias = str | bytes | Mapping[str, Any] | Iterable[tuple[str, str | None]] | IO[Any] | None
-
+_Data: TypeAlias = str | bytes | Mapping[str, Any] | Iterable[tuple[str, str | None]] | IO[Any]
+_Auth: TypeAlias = Union[tuple[str, str], _auth.AuthBase, Callable[[PreparedRequest], PreparedRequest]]
+_Cert: TypeAlias = Union[str, tuple[str, str]]
```

Comment:
Should this accept bytes? `adapters.pyi` currently does this:

https://github.com/python/typeshed/blob/54c8f45869a67ad2fc73a0d59ce316dd723a2e94/stubs/requests/requests/adapters.pyi#L47

Looking at `sessions.py`, it seems to just pass everything to the adapter, so if adapters work with bytes, then we should allow bytes here too. In `adapters.py`, the certs are validated with `os.path.exists`, which works with bytes and strings.

In general, `adapters.pyi` seems to be a bit out of sync with `api.pyi` and `sessions.pyi`, but I don't think fixing it belongs to this PR.

Reply:
Looks like it would work for `requests`, but in the end it ends up at urllib3's `HTTPSConnectionPool`. Not sure if that accepts & keeps accepting `bytes`.  Their current version doesn't have type stubs, but their [upcoming major version](https://urllib3.readthedocs.io/en/latest/reference/urllib3.connectionpool.html#urllib3.HTTPSConnectionPool) requires `str`.

I can make a follow-up PR later to work away some of the inconsistencies and take a look at `models`.